aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -526,11 +526,9 @@ std::optional<Tensor> convert_boolean_attn_mask(const std::optional<Tensor>& att
   // Convert boolean mask to additive mask; need to invert mask to indicate what
   // to mask *out*.
   if (attn_mask->dtype() == at::kBool) {
-    auto new_attn_mask = at::zeros_like(attn_mask.value(), dtype);
     // TODO Use the max type of the input and output
-    new_attn_mask.masked_fill_(
-        attn_mask->logical_not(), -std::numeric_limits<double>::infinity());
-    return new_attn_mask;
+    auto new_attn_mask = at::zeros_like(attn_mask.value(), dtype);
+    return at::where(attn_mask->logical_not(), -std::numeric_limits<double>::infinity(), new_attn_mask);
```

Comment:
Does `at::where(attn_mask->logical_not(),-std::numeric_limits<double>::infinity(), 0.)` work? As written we are allocating two tensors; we should be able to get this to allocate just one tensor.